src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterServiceMonitor.scala

```diff
@@ -0,0 +1,105 @@
+
+object ClusterServiceMonitor {
+
+
+  val newRelic: NewRelicMetrics = Metrics.newRelic
+
+  def props(config: ClusterServiceConfig, gdDAO: GoogleDataprocDAO, googleProjectDAO: GoogleProjectDAO, dbRef: DbReference, welderDAO: WelderDAO, jupyterDAO: JupyterDAO, newRelic: NewRelicMetrics): Props = {
+    Props(new ClusterServiceMonitor(config, gdDAO, googleProjectDAO, dbRef, welderDAO, jupyterDAO, newRelic: NewRelicMetrics))
+  }
+
+  sealed trait ClusterServiceMonitorMessage
+  case object DetectClusterStatus extends ClusterServiceMonitorMessage
+  case object TimerKey extends ClusterServiceMonitorMessage
+
+  sealed trait ServiceStatus extends Product with Serializable
+  sealed trait JupyterStatus extends ServiceStatus with Product with Serializable
+  object JupyterStatus {
+    final case object JupyterOK extends JupyterStatus
+    final case object JupyterDown extends JupyterStatus
+  }
+
+  sealed trait WelderStatus extends ServiceStatus with Product with Serializable
+  object WelderStatus {
+    final case object WelderOK extends WelderStatus
+    final case object WelderDown extends WelderStatus
+  }
+
+  case class Status(val welderStatus: WelderStatus, val jupyterStatus: JupyterStatus)
+}
+
+/**
+  * This monitor periodically sweeps the Leo database and checks for clusters which no longer exist in Google.
+  */
+class ClusterServiceMonitor(config: ClusterServiceConfig, gdDAO: GoogleDataprocDAO, googleProjectDAO: GoogleProjectDAO, dbRef: DbReference, welderDAO: WelderDAO, jupyterDAO: JupyterDAO, newRelic: NewRelicMetrics) extends Actor with Timers with LazyLogging {
+
+  import context._
+
+  override def preStart(): Unit = {
+    super.preStart()
+    timers.startPeriodicTimer(TimerKey, DetectClusterStatus, config.pollPeriod)
+  }
+
+  override def receive: Receive = {
+    case DetectClusterStatus =>
+      // Get active clusters from the Leo DB, grouped by project
+      val activeClusters: Future[List[Cluster]] = getActiveClustersFromDatabase.flatMap(clusterMap => {
+        Future.successful(clusterMap.values.flatten.toList)
+      })
+
+      //check the status of the workers in each active cluster and log instances of a down worker to new relic
+      activeClusters.foreach { cs =>
+        cs.foreach { cluster =>
+          checkClusterStatus(cluster).foreach { status =>
+
+            if (status.jupyterStatus.equals(JupyterStatus.JupyterDown)) {
+              logger.info("jupyter down")
+              newRelic.incrementCounterIO("jupyterDown").unsafeRunAsync(_ => ())
+            }
+
+            if (status.welderStatus.equals(WelderStatus.WelderDown)) {
+              logger.info("welder enabled and down")
+              newRelic.incrementCounterIO("welderDown").unsafeRunAsync(_ => ())
+            }
+
+          ()
+          }
+        }
+      }
+  }
+
+  private def getActiveClustersFromDatabase: Future[Map[GoogleProject, Seq[Cluster]]] = {
+    dbRef.inTransaction {
+      _.clusterQuery.listActive
```

Comment:
I think this query joins to `CLUSTER_LABEL` which is not needed here - could we make a `listActiveClusterOnly` method which avoids the join?